src/query/queryWithCursor.ts

```diff
@@ -39,15 +39,36 @@ export async function queryWithCursor<T extends AnyObject>(
     table.cursorSecret,
   );
 
-  const {
-    LastEvaluatedKey,
-    Items,
-    ScannedCount,
-  }: QueryOutput = await dbClient.query(params).promise();
+  let itemsData = [];
+  let LastEvaluatedKeyOutput: Key = undefined;
+  let canIterate = true;
+  let scannedCountCopy = 0;
+
+  do {
+    const {
+      LastEvaluatedKey,
+      Items,
+      ScannedCount,
+    }: QueryOutput = await dbClient.query(params).promise();
+
+    LastEvaluatedKeyOutput = LastEvaluatedKey;
+
+    scannedCountCopy = ScannedCount;
```

Comment:
you need to increment this every time this loops run

Reply:
Changed by lakhmaji